### ISSUE TITLE
ESP32-C3 pin glitches during start-up

### ISSUE THREAD

**Issue by martinjaeger** [zephyrproject-rtos/zephyr#51250]
**Describe the bug**

The pin levels of the ESP32-C3 don't seem to be at a defined state during start-up and show some glitches. I realized this while debugging an I2C application, where this seems to confuse other participants on the bus, so that it corrupts the communication.

Below image shows the start-up for pins `I2C0_SDA_GPIO6` (channel 1) and `I2C0_SCL_GPIO7` (channel 0).

Both pins have an external 10k pull-up resistor to 3V3.

![image](https://user-images.githubusercontent.com/17674105/195572607-a8a11890-ec16-4b06-a19c-d333189bdfeb.png)

I pressed reset 3 times. The first time, both pins show a small spike where they seem to drive the bus low. The second time GPIO6 is kept low until the next press of the reset button.

The behaviour seems totally random. In most cases you get the small spike (pin driven low).

**To Reproduce**

I have initally tested this with my custom application, but it occurs even for hello world, where the I2C peripheral is not used.

`west build -b esp32c3_devkitm samples/hello_world`

Add an external pull-up (10k in my case) to above mentioned pins.

Connect a logic analyzer / oscilloscope to the pins and press reset a couple of times.

**Expected behavior**

The pins should be configured as analog inputs during startup.

**Impact**

Unpredictable pin behavior can accidentally destroy hardware (e.g. in case of power electronics, where a MOSFET is switched on).

**Environment (please complete the following information):**

 - OS: Linux
 - Toolchain: Zephyr
 - Commit SHA or Version used: cfcf63018481a62fb413859555fb2e8e14d536c7

**Comment by sylvioalves** [comment]
Hi @martinjaeger, I think isn't an code issue. It is HW behavior only.  
If you check [ESP32-C3 datasheet](https://www.espressif.com/sites/default/files/documentation/esp32-c3_datasheet_en.pdf), page 21-22 and Table-7, you can see that a few pins have low-level glitch during power up.  
Check pins MTCK (GPIO6) and MTDO (GPIO7):

![image](https://user-images.githubusercontent.com/12415528/195581003-ff0c62a8-8f16-4a97-9968-6c0deccc9962.png)

If the glitch you noticed isn't that, then we need to fix it.

**Comment by martinjaeger** [comment]
Thanks @sylvioalves for the fast response! Looks like I had bad luck with the choice of the pins ;)

But I think it's a different kind of glitch.
1. The pins sometimes stay low (as for the second reset above).
2. The glitch is orders of magnitude longer (380 µs, see close-up below).

![image](https://user-images.githubusercontent.com/17674105/195583304-8c277ebe-88e0-4857-8562-0da76e0a5382.png)

**Comment by sylvioalves** [comment]
Oh, this is way longer than the HW glitch. Can you confirm whether this is from I2C code startup?

**Comment by sylvioalves** [comment]
@martinjaeger, I have tried with `hello_world` sample code and no issues so far. Any suggestion?

**Comment by martinjaeger** [comment]
Interesting. I tried with a different board (same type, also `esp32c3_devkitm`) and could not see the glitches with `hello_world`. Afterwards I flashed my custom firmware, where I could see the glitches again. Now if I flash `hello_world` again, the glitches are still there. Even after `esptool.py erase_flash`, disconnect from power and with no firmware on it, I can see the glitches. The serial prints `invalid header: 0xffffffff` all the time, which I guess is from the bootloader which doesn't find an image.

Any idea how I can fully reset the board in addition to the steps above? I'd like to try to find an example that introduces the glitches, but first have to get rid of them again ;)
